Sources/Brave/Frontend/Sync/BraveCore/BraveSyncAPIExtensions.swift

```diff
@@ -96,11 +107,18 @@ extension BraveSyncAPI {
     }
   }
 
-  func addServiceStateObserver(_ observer: @escaping () -> Void) -> AnyObject {
+  /// Method to add observer for SyncService for onStateChanged and onSyncShutdown
+  /// OnStateChanged can be called in various situations like successful initialization - services unavaiable
+  /// sync shutdown - sync errors - sync chain deleted
+  /// - Parameters:
+  ///   - onStateChanged: Callback for sync service state changes
+  ///   - onServiceShutdown: Callback for sync service shutdown
+  /// - Returns: Listener for service
+  func addServiceStateObserver(_ onStateChanged: @escaping () -> Void, onServiceShutdown: @escaping () -> Void) -> AnyObject {
```

Comment:
It seems like the shutdown arg is not provided a lot, maybe we could make it default to an empty closure? 

i.e.

```swift
func addServiceStateObserver(
  _ onStateChanged: @escaping () -> Void, 
  onServiceShutdown: @escaping () -> Void = { }
) -> AnyObject {
  ...
}
```

Reply:
Yes I  totally agree, missed it, to many changes about this part Fixed at https://github.com/brave/brave-ios/pull/7031/commits/3d172a29f0a8d62dacb9675594e1f0f30ae29ee7